[BUG] fix `_update` default for late `fh` pass case

fkiraly opened the pull request:
Fixes https://github.com/alan-turing-institute/sktime/issues/2361:

If no `fh` was passed in `fit`, and `update` was called with the default `_update`, it would break as it tried to read `self.fh` which would raise an error. Instead, `self_fh` should be read, which does not raise an error.

Lovkush-A commented:
Looks fine to me. Do we need a unittest to make sure this bug does not occur again?

Lovkush-A reviewed (changes requested):
Just reading base class to check what self._fh is. I noticed the docstring for _update is inaccurate. It does not have entry for update_params when it needs one, and it has entry for fh when fh is not a parameter.

fkiraly commented:
Yes, there is more wrong with the docstrings, too.
I went over all of them and updated them here:
https://github.com/alan-turing-institute/sktime/pull/2314

I updated `_update` here - it was correct in the extension template, but not in the base class.

Lovkush-A reviewed (approved):
looks fine to me. replacing self.fh with self._fh should prevent inapropriate raising errors when self._fh is None.

fkiraly commented:
passed already before change to docstrings only; readthedocs errors out with 0sec, that´s most likely a connection error?
